app/controllers/api/v1/stories_controller.rb

```diff
@@ -2,13 +2,31 @@ module Api
   module V1
     class StoriesController < ApiController
       def index
-        stories = Story.limit(stories_limit).order(:published_at)
-
+        stories = stories_filter(params[:tags])
         render json: stories, each_serializer: Api::V1::StorySerializer
       end
 
       private
 
+      def stories_filter(tags)
+        tagged_stories = tagged_stories(tags)
+        return tagged_stories if tagged_stories.length >= 5
+        more_stories = untagged_stories
+        return (tagged_stories + more_stories).first(5) if tagged_stories
+        more_stories
+      end
+
+      def tagged_stories(tags)
```

Comment:
I think that these methods should go into the model, instead of the controller.
And can you add some unit specs for these? (tagged_stories and untagged_stories)